packages/build/src/time/report.ts

```diff
@@ -1,29 +1,73 @@
-import { closeClient, startClient } from '../report/statsd.js'
+import type { Tags, StatsD } from 'hot-shots'
+
+import { closeClient, formatTags, startClient } from '../report/statsd.js'
 
 import { addAggregatedTimers } from './aggregate.js'
 import { roundTimerToMillisecs } from './measure.js'
 
+interface Timer {
+  metricName: string
+  stageTag: string
+  parentTag: string
+  durationNs: number
+  tags: Record<string, string | string[]>
+}
+
+// TODO Once we have more TS coverage make host & port required and instead make `statsdOpts` below optional
+// that should be more clean
+export interface StatsDOptions {
+  host?: string
+  port?: number
+}
+
+interface TimerOptions {
+  timers: Timer[]
+  statsdOpts: StatsDOptions
+  framework?: string
+}
+
 // Record the duration of a build phase, for monitoring.
 // Sends to statsd daemon.
-export const reportTimers = async function ({ timers, statsdOpts: { host, port }, framework }) {
-  if (host === undefined) {
+export const reportTimers = async function ({
+  timers,
+  statsdOpts: { host, port },
+  framework,
+}: TimerOptions): Promise<void> {
+  if (host === undefined || port === undefined) {
     return
   }
 
   const timersA = addAggregatedTimers(timers)
-  await sendTimers({ timers: timersA, host, port, framework })
+  await sendTimers({ timers: timersA, statsdOpts: { host, port }, framework })
 }
 
-const sendTimers = async function ({ timers, host, port, framework }) {
+const sendTimers = async function ({
+  timers,
+  statsdOpts: { host, port },
+  framework,
+}: TimerOptions & { statsdOpts: Required<StatsDOptions> }): Promise<void> {
   const client = await startClient(host, port)
   timers.forEach((timer) => {
     sendTimer({ timer, client, framework })
   })
   await closeClient(client)
 }
 
-const sendTimer = function ({ timer: { metricName, stageTag, parentTag, durationNs, tags }, client, framework }) {
+const sendTimer = function ({
+  timer: { metricName, stageTag, parentTag, durationNs, tags },
+  client,
+  framework,
+}: {
```

Comment:
NIT: can we get rid of the object destruction as typing makes it super hard to read?

whats wrong with good old params? They won't change in the future and it's only 3 of them but 10 times easier to read and type

```suggestion
const sendTimer = function (
  timer: Timer,
  client: StatsD,
  framework?: string
```